foundry/foundry.py

```diff
@@ -529,6 +538,23 @@ def download(self, globus=True, verbose=False, **kwargs):
                  "grouper": "matio"
                 }
             xtract_https_download(self, verbose=verbose, **xtract_config)
+
+        # after download check making sure directory exists, contains proper amount of files
+        if os.path.isdir(path):
+            #checking for proper number of files downloaded
+            dir_length = len(os.listdir(path))
+            if self.dataset.splits:
+                if(dir_length != len(self.dataset.splits)):
+                    print("Error: Incorrect number of files in directory")
```

Comment:
Excellent, I would just slightly adjust this so you're printing in the exception, like so: https://stackoverflow.com/questions/2052390/manually-raising-throwing-an-exception-in-python